Incorrect calls to pony_asio_event_resubscribe_read/write with Disposable/Destroyed event

See https://circleci.com/gh/ponylang/ponyc/5086?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

In an entirely unrelated PR, process.Expect failed because of an assertion thrown in `pony_asio_event_resubscribe_read`. That shouldn't happen unless there is a race condition.

We need to figure out how the race condition could happen based on that test and how to prevent it.

Based on the backtrace, this call came from `ponyint_asio_backend_dispatch` where `ev` had to be non-null. So it was either a dispose or destroy event.

This block

```c
      if(flags != 0)
      {
        if(ev->auto_resub && !(flags & ASIO_WRITE))
          pony_asio_event_resubscribe_write(ev);
        if(ev->auto_resub && !(flags & ASIO_READ))
          pony_asio_event_resubscribe_read(ev);
        pony_asio_event_send(ev, flags, count);
      }
```

from epoll.c that starts at line 275 should not be executed if flags ASIO_DISPOSABLE or ASIO_DESTROYED are set.

This happened to me in https://circleci.com/gh/winksaville/ponyc/1060, the backtrace was:
```
245 tests started, 244 complete: process/STDIN-STDOUT started
245 tests started, 245 complete: process/STDIN-STDOUT complete
src/libponyrt/asio/epoll.c:181: pony_asio_event_resubscribe_read: Assertion `0` failed.

246 tests started, 245 complete: process/STDERR started
Backtrace:
  ./stdlib(ponyint_assert_fail+0xf1) [0x8a8801]
  ./stdlib(pony_asio_event_resubscribe_read+0x58) [0x8b2ac4]
  ./stdlib(ponyint_asio_backend_dispatch+0x2cd) [0x8b2e50]
  /lib/x86_64-linux-gnu/libpthread.so.0(+0x76ba) [0x7feb90ac56ba]
  /lib/x86_64-linux-gnu/libc.so.6(clone+0x6d) [0x7feb900d841d]
Makefile:923: recipe for target 'test-ci' failed
make[1]: *** [test-ci] Aborted
make[1]: Leaving directory '/home/pony/project'
Makefile-lib-llvm:27: recipe for target 'test-ci' failed
make: *** [test-ci] Error 2
Exited with code 2
```
I searched for pony_asio_event_resubscribe_read and see calls to it from src/libponyrt/asio/epoll.c as well as packages/net/tcp_connection.pony which, on the face of it, means that calls to pony_asio_event_* could be from multiple threads and hence there seems likily there would be some race conditions:
```
$ ack pony_asio_event_resubscribe_read
packages/net/tcp_connection.pony
7:use @pony_asio_event_resubscribe_read[None](event: AsioEventID)
805:            @pony_asio_event_resubscribe_read(_event)

src/libponyrt/asio/epoll.c
175:PONY_API void pony_asio_event_resubscribe_read(asio_event_t* ev)
280:          pony_asio_event_resubscribe_read(ev);

src/libponyrt/asio/kqueue.c
102:PONY_API void pony_asio_event_resubscribe_read(asio_event_t* ev)
```
On the face of it, if calling pony_asio_event_* from multiple threads is valid, than maybe we are being to stringent in epoll.c and instead of asserting we should just ignore the call. So maybe this conditional statement we find sprinkled in epoll.c:
```
  if((ev == NULL) ||
    (ev->flags == ASIO_DISPOSABLE) ||
    (ev->flags == ASIO_DESTROYED))
  {
    pony_assert(0);
    return;
  }
```
Should be two conditional statements:
```
  if((ev == NULL) {
    pony_assert(0);
    return;
  }
  if((ev->flags == ASIO_DISPOSABLE) ||
    (ev->flags == ASIO_DESTROYED))
  {
    return;
  }
```
Or maybe just remove the assert all together.

--- a/src/libponyrt/asio/epoll.c
+++ b/src/libponyrt/asio/epoll.c
@@ -274,10 +274,13 @@ DECLARE_THREAD_FN(ponyint_asio_backend_dispatch)
 
       if(flags != 0)
       {
-        if(ev->auto_resub && !(flags & ASIO_WRITE))
-          pony_asio_event_resubscribe_write(ev);
-        if(ev->auto_resub && !(flags & ASIO_READ))
-          pony_asio_event_resubscribe_read(ev);
+        if (!(flags & ASIO_DESTROYED) || !(flags & ASIO_DISPOSABLE))
+        {
+          if(ev->auto_resub && !(flags & ASIO_WRITE))
+            pony_asio_event_resubscribe_write(ev);
+          if(ev->auto_resub && !(flags & ASIO_READ))
+            pony_asio_event_resubscribe_read(ev);
+        }
         pony_asio_event_send(ev, flags, count);
       }
     }

